Gems/Atom/Utils/Code/Source/ImageComparison.cpp

```diff
@@ -9,11 +9,55 @@
 #include <Atom/Utils/ImageComparison.h>
 
 #include <AzCore/Casting/numeric_cast.h>
+#include <AzCore/Preprocessor/EnumReflectUtils.h>
 
 namespace AZ
 {
     namespace Utils
     {
+        AZ_ENUM_DEFINE_REFLECT_UTILITIES(ImageDiffResultCode);
+
+        void ImageDiffResult::Reflect(ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+            {
+                ImageDiffResultCodeReflect(*serializeContext);
+
+                serializeContext->Class<ImageDiffResult>()
+                    ->Version(1)
+                    ->Field("ResultCode", &ImageDiffResult::m_resultCode)
+                    ->Field("DiffScore", &ImageDiffResult::m_diffScore)
+                    ->Field("FilteredDiffScore", &ImageDiffResult::m_filteredDiffScore);
+            }
+
+            if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+            {
+                behaviorContext->EnumProperty<static_cast<int>(ImageDiffResultCode::Success)>("ImageDiffResultCode_Success")
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                    ->Attribute(AZ::Script::Attributes::Module, "utils");
+                behaviorContext->EnumProperty<static_cast<int>(ImageDiffResultCode::SizeMismatch)>("ImageDiffResultCode_SizeMismatch")
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                    ->Attribute(AZ::Script::Attributes::Module, "utils");
+                behaviorContext->EnumProperty<static_cast<int>(ImageDiffResultCode::FormatMismatch)>("ImageDiffResultCode_FormatMismatch")
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                    ->Attribute(AZ::Script::Attributes::Module, "utils");
+                behaviorContext->EnumProperty<static_cast<int>(ImageDiffResultCode::UnsupportedFormat)>("ImageDiffResultCode_UnsupportedFormat")
+                    ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                    ->Attribute(AZ::Script::Attributes::Module, "utils");
```

Comment:
AZ_ENUM macros produce a function that you can call and set this up automatically

Reply:
Per our offline discussion, this is still needed for exposing the enum to pytest, before we have more support from AZ_ENUM_DEFINE_REFLECT_UTILITIES.